tools/mkmapdata/mkmapdata.sh

```diff
@@ -530,7 +532,23 @@ until ((nfile>${#INGRID[*]})); do
       runcmd "ncatted -a history,global,a,c,"$history"  ${OUTFILE[nfile]}"
       runcmd "ncatted -a hostname,global,a,c,$HOST   -h ${OUTFILE[nfile]}"
       runcmd "ncatted -a logname,global,a,c,$LOGNAME -h ${OUTFILE[nfile]}"
+   fi
 
+   nfile=nfile+1
+done
+
+intelvers=18.0.5
+module purge
+module load intel/$intelvers
+module load ncl
+
+declare -i nfile=1
+until ((nfile>${#INGRID[*]})); do
+
+   # Skip if large file and Fast mode is on
+   if [ "$fast" = "YES" ] && [ "${SRC_LRGFIL[nfile]}" = "netcdf4" ]; then
+      echo "Skipping ${OUTFILE[nfile]} as fast mode is on so skipping large files in NetCDF4 format"
+   else
```

Comment:
This set of changes feels fragile, both in that you are doing some different module loads (without a machine-specific conditional), and in that you are repeating some logic from above. I think both of these things are going to make this hard to maintain. 

But here's the probably good news: I was just looking back at the history of this rmdups script. It looks like we introduced it in Fall, 2011 to deal with problems with the then-current ESMF remapper (version 5.2). Bob Oehmke apparently fixed this in a version put out in Nov, 2011 (`ESMF_5_2_0rp1_beta_snapshot_14`). So I strongly suspect that we don't need this rmdups script any more.

Do you have output from all of these rmdups calls that you could use to check if it was actually doing anything? If so, could you look for whether any files generated the message, `Successfully removed duplicate weights` as opposed to `No duplicate weights to remove`? Alternatively (if the output from that script isn't saved), you could regenerate all mapping files with this code removed, and confirm that they are all identical to the ones you generated before.

Note that, as of 2011, the problematic files were all spectral element grids (ne120np4 and ne30np4). So those would be the ones to especially keep an eye on, but if it's not much harder to check this for all grids, that would be good.

I know this request adds even more time to an already painful set of changes. However, I think this is worth doing because, for the project to rework the toolchain, it will be good to know if we still need this rmdups functionality for any reason. (But it's okay with me if you want to bring this in as it currently is, then address this comment in a follow-up.)

Reply:
According to my regrid.o output files, this is the only case where rmdups.ncl removes duplicates, but then it runs into a problem. Pls let me know what you think, @billsacks:

(0)     map file: map_1km-merge-10min_HYDRO1K-merge-nomask_to_3x3min_nomask_aave_da_c200206.nc
(0)     n_s = 83822600 max(row)=25008596 max(col)=125599448
(0)     checking for dups, sorting...
(0)     zeroing out any dups...
(0)     ndups = 14163
(0)     compacting S...
(0)     removed 14163 dups
(0)     writing new netcdf file
warning:_NclOpenFile: cannot open file <rmdups_map_1km-merge-10min_HYDRO1K-merge-nomask_to_3x3min_nomask_aave_da_c200206.nc>; No such file or directory

^Mwarning:Attempt to reference attribute (history) which is undefined
^Mfatal:["Execute.c":8637]:Execute: Error occurred at or near line 115 in file ./rmdups.ncl

Reply:
Thanks for checking. Is that even a mapping file that we use? I don't see anything like that (or any mapping files to 3 min resolution) in the xml file. If that's not even used, and if that's the only instance where the duplicate-removal is happening, I think we're safe to remove rmdups from our workflow. @ekluzek let us know if you object to this.

(My sense from the 2011 emails is that, even if there are duplicates, they probably aren't doing much harm, other than making files a bit bigger than they need to be.)

Reply:
The rmdups.ncl code was important for removing duplicates in the past versions of ESMF. And was only important in making things faster. So if it's only finding duplicates for maps we aren't using, removing it saves complexity and makes the process more maintainable. And the only downside is in performance if there happens to be a map that ESMF makes with duplicates. That's what I'm thinking anyway. So yeah I'd say go ahead with that change.